video/out/gpu/video.c

```diff
@@ -1291,8 +1299,22 @@ static void copy_image(struct gl_video *p, int *offset, struct image img)
         img.multiplier *= 1.0 / (tex_max - 1);
     }
 
-    GLSLF("color.%s = %f * vec4(texture(texture%d, texcoord%d)).%s;\n",
-          dst, img.multiplier, id, id, src);
+    if (img.interlaced) {
+        // Simple "blend" deinterlacer based on vlc's deinterlace/algo_basic.c
+        // The two fields are averaged together to produce one frame.
+        // Each line output is the input line averaged with the previous line.
+        // The very first line is copied exactly (albeit by averaging itself).
+        GLSLF("float yprev%d = texcoord%d.y == 0.0 ? 0.0 : "
+                              "texcoord%d.y - 1.0 / texture_size%d.y;\n",
+              id, id, id, id);
+        GLSLF("vec2 prev%d = vec2(texcoord%d.x, yprev%d);\n", id, id, id);
+        GLSLF("color.%s = %f * mix(texture(texture%d, texcoord%d), "
+                                  "texture(texture%d, prev%d), 0.5).%s;\n",
```

Comment:
If the texture filter is linear, this can (and should) be simplified to a single sample (of the position exactly in between `texcoord%d.y` and `prev%d`).